paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java

```diff
@@ -801,6 +805,50 @@ public boolean dropItem(boolean dropAll) {
         // Paper end - Fix HumanEntity#drop not updating the client inv
     }
 
+    @Override
+    public @Nullable Item dropItem(final int slot, final int amount) {
+        if (slot < 0 || slot >= this.inventory.getSize()) {
+            throw new IllegalArgumentException("Slot " + slot + " is not a valid inventory slot.");
+        }
+
+        return dropItemRaw(this.inventory.getItem(slot), amount, false);
+    }
+
+    @Override
+    public @Nullable Item dropItem(final @NotNull EquipmentSlot slot, final int amount) {
+        return dropItemRaw(this.inventory.getItem(slot), amount, false);
+    }
+
+    @Override
+    public @Nullable Item dropItemRandomly(final int slot, final int amount) {
+        if (slot < 0 || slot >= this.inventory.getSize()) {
+            throw new IllegalArgumentException("Slot " + slot + " is not a valid inventory slot.");
+        }
+
+        return dropItemRaw(this.inventory.getItem(slot), amount, true);
+    }
+
+    @Override
+    public @Nullable Item dropItemRandomly(final @NotNull EquipmentSlot slot, final int amount) {
+        return dropItemRaw(this.inventory.getItem(slot), amount, true);
+    }
+
+    private @Nullable Item dropItemRaw(final ItemStack originalItemStack, final int amount, final boolean throwRandomly) {
+        if (originalItemStack == null || originalItemStack.isEmpty() || amount <= 0) {
+            return null;
+        }
+
+        final net.minecraft.world.item.ItemStack nmsItemStack = CraftItemStack.unwrap(originalItemStack);
+        final net.minecraft.world.item.ItemStack dropContent = nmsItemStack.split(Math.min(originalItemStack.getAmount(), amount));
```

Comment:
I don't think you need to do the Math.min. That should be part of the nmsItemStack.split now.